packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -82,7 +92,17 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
   void _handlePointerDataPacket(ui.PointerDataPacket packet) {
     // We convert pointer data to logical pixels so that e.g. the touch slop can be
     // defined in a device-independent manner.
-    _pendingPointerEvents.addAll(PointerEventConverter.expand(packet.data, window.devicePixelRatio));
+    final Iterable<PointerEvent> events =
+        PointerEventConverter.expand(packet.data, window.devicePixelRatio);
+    if (resamplingEnabled) {
+      for (final PointerEvent event in events) {
+        final PointerEventResampler resampler = _resamplers.putIfAbsent(
+            event.device, () => PointerEventResampler());
```

Comment:
Should we try to remove the resampler when an up/remove event happens? I'm not sure if some OS would generate a very large number of different `event.device` that would eventually make this look like a memory leak.

Reply:
Yes, I'm removing resamplers that are inactive in latest change.